Update password + hide/unhide password

G-Ray opened the pull request.

G-Ray committed:
TextField: Add a button prop

G-Ray committed:
Password Edit modal

G-Ray committed:
hide/unhide password

G-Ray committed:
hide/unhide password on login page

sujeebant commented:
- When the secret key is long enough , the 'eye icon' is above the secret key.
- I think there is no need for the possibility to see the secret key while logging in, it is useful for register and secret key updates.
- Aren't the two icons reversed ? I think the eye should be displayed when secret key is visible, like in this animation https://twitter.com/crafty/status/767767935255453696

G-Ray committed:
textinput password: fix padding-right

G-Ray committed:
TextField: hide/unhide icon only when length > 0

G-Ray commented:
> * When the secret key is long enough , the 'eye icon' is above the secret key.

Fixed

The behavior is confirmed by the UX designer. Also, I updated the code to only display the hide/unhide icon when the password length is > 0.

leventdem reviewed (approved):
ok
